src/deployCommand.ts

```diff
@@ -162,6 +179,87 @@ export abstract class DeployCommand extends SourceCommand {
 
     return this.isAsync ? this.report(validatedDeployId) : this.poll(validatedDeployId);
   }
+
+  protected createRequestedReports(): void {
+    if (this.flags.coverageformatters) {
+      this.createCoverageReport(this.deployResult, this.flags.coverageformatters, 'no-map', this.flags.outputdir);
+    }
+    if (this.flags.junit && !this.isAsync) {
+      this.createJunitResults(this.deployResult);
+    }
+  }
+
+  protected createCoverageReport(
+    deployResult: DeployResult,
+    formatters: string[],
+    sourceDir: string,
+    outputDir: string
+  ): void {
+    const apexCoverage = transformCoverageToApexCoverage(
+      toArray(deployResult.response?.details?.runTestResult?.codeCoverage)
+    );
+    fs.mkdirSync(outputDir, { recursive: true });
+    const options = this.getCoverageFormattersOptions(formatters);
+    const coverageReport = new CoverageReporter(apexCoverage, outputDir, sourceDir, options);
+    coverageReport.generateReports();
+  }
+
+  protected getCoverageFormattersOptions(formatters: string[] = []): CoverageReporterOptions {
+    /* eslint-disable @typescript-eslint/no-unsafe-member-access,@typescript-eslint/no-unsafe-assignment */
+    const options: CoverageReporterOptions = {} as CoverageReporterOptions;
+    // set requested report formats
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    options.reportFormats = formatters.map((format) => format as CoverageReportFormats);
```

Comment:
I don't think you need to use `map` here to cast the type. You can just do

```typescript
options.reportFormats = formatters as CoverageReportFormats[]
```

Actually, couldn't you change the method signature to take `formatters` as `CoverageReportFormats[]` instead of `string[]`?

Reply:
Moving up into signature forces the cast to the caller. Would rather handle that need in the function.